### PR TITLE
Future for module import bug with `search.chpl`

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
Adds a new future to track the bug described in https://github.com/chapel-lang/chapel/issues/23100.

Adjusts the tests in `types/string/psahabu/perf/` to work around the error that we were seeing on Mac testing.
- renames `search.chpl` to `stringSearch.chpl`
- keeps the same name for the `.graph`, `.perfkeys`, and `.dat` files to avoid disrupting nightly testing data

**Testing:**
- [x] paratest
- [x] future matching `.bad` on Mac OS
- [x] string perf tests passing with `-performance` flag

### CODE REVIEW

File: test/modules/corrado/import-ambiguity-bug/otherModule.future [hunk 1]
```diff
@@ -0,0 +1,4 @@
+compiling a file adjacent to a module named "search.chpl" when importing the
+Time module causes an import ambiguity error and an internal compiler error
+
+See #23100
```
Comment (lydia-duncan) [inline test/modules/corrado/import-ambiguity-bug/otherModule.future:4]: This doesn't use our normal future formatting.  Typically we start with:
```
<category>: short description
#<issue number>

optional longer description
```
https://chapel-lang.org/docs/latest/developer/bestPractices/TestSystem.html#a-test-that-tracks-a-failure states this and test/functions/ferguson/hijacking/issue-18081.future has an actual example of it.

File: test/types/string/psahabu/perf/stringSearch.perfexecopts [hunk 2]
```diff
@@ -0,0 +1 @@
+--bogus=true # search.perfkeys
```
Comment (lydia-duncan) [inline test/types/string/psahabu/perf/stringSearch.perfexecopts:1]: Is an additional flag necessary?  I though it was sufficient to have a single space followed by the `# search.perfkeys`

**Review by lydia-duncan — COMMENTED** [review]
Couple of nits, thanks for fixing this!

File: test/types/string/psahabu/perf/stringSearch.perfexecopts (hunk 2 above)
Comment (jeremiah-corrado) [reply test/types/string/psahabu/perf/stringSearch.perfexecopts:1]: Unfortunately that wasn't working :(

Comment (jeremiah-corrado) [reply test/types/string/psahabu/perf/stringSearch.perfexecopts:1]: Nvm, I was missing the space. I'll fix it.

**Review by lydia-duncan — APPROVED** [review]
Looks great, thanks!

**Pull request merged**